src/TorchVision/Functional.cs

```diff
@@ -24,6 +24,19 @@ public static partial class transforms
         {
             public static partial class functional
             {
+                /// <summary>
+                /// Get the image dimensions
+                /// </summary>
+                /// <param name="img">An image tensor.</param>
+                /// <returns>The number of channels, the height and width of the image</returns>
+                public static (long channels, long height, long width) get_dimensions(Tensor img)
+                {
+                    var d = img.Dimensions;
+                    var channels = d == 2 ? 1 : img.shape[d - 3];
```

Comment:
Can it ever be NxHxW, or is it always Nx1xHxW?

Reply:
In PyTorch, this convention is NxCxHxW. so NxHxW, does not fit that convention but Nx1xHxW does.
Anyway implementation is from here:
https://github.com/pytorch/vision/blob/498b9c8662e2322615748aafc321ad4a5bc02afb/torchvision/transforms/v2/functional/_meta.py#L14